syft/pkg/cataloger/java/parse_gradle_lockfile.go

```diff
@@ -0,0 +1,64 @@
+package java
+
+import (
+	"bufio"
+	"strings"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/anchore/syft/syft/source"
+)
+
+const gradleLockfileDirGlob = "**/gradle.lockfile*"
+
+// Dependency represents a single dependency in the gradle.lockfile file
+type LockfileDependency struct {
+	Group   string
+	Name    string
+	Version string
+}
+
+func parserGradleLockfile(_ source.FileResolver, _ *generic.Environment, reader source.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	var pkgs []pkg.Package
+
+	// Create a new scanner to read the file
+	scanner := bufio.NewScanner(reader)
+
+	// Create slices to hold the dependencies and plugins
+	dependencies := []LockfileDependency{}
+
+	// Loop over all lines in the file
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Trim leading and trailing whitespace from the line
+		line = strings.TrimSpace(line)
+
+		groupNameVersion := line
+		groupNameVersion = strings.Trim(groupNameVersion, "\"")
+		parts := strings.Split(groupNameVersion, ":")
+
+		// we have a version directly specified
+		if len(parts) == 3 {
+			version := strings.Split(parts[2], "=")
+			// Create a new Dependency struct and add it to the dependencies slice
```

Comment:
i'm not quite sure if I understand this right, because you only switch from equally to greater than 3 so this wouldn't match anymore because its only 3 parts? 

Or do you mean expecting it to have always three parts and just handling some type of exception when there are more?

Reply:
I mean: from what I see in the format examples it's `<group,name,version>=<stuff>` -- it seems more clear to me if we get rid of the `=<stuff>` and then just deal directly with the `<group,name,version>` part

Reply:
ahh yeah youre totally right, thats still from the gradle and not the lockfile stuff..